AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -140,18 +139,21 @@ + (MSTokenResult *)retrieveCachedToken:(NSString *)partitionName {
     NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partitionName]];
     if (tokenString) {
       MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:tokenString];
+      if (tokenResult && ![tokenResult.status isEqualToString:kMSTokenResultSucceed]) {
```

Comment:
You can call a selector on nil and it will return NO here.
```suggestion
      if (![tokenResult.status isEqualToString:kMSTokenResultSucceed]) {
```

Reply:
`tokenResult` is just instantiated so `tokenResult` won't be `nil`. Redundant check.
```suggestion
      if (![kMSTokenResultSucceed isEqualToString:tokenResult.status]) {
```
If you use the constant to compare the string, it won't have the issue that @guperrot mentioned above.